lib/guest_configuration_base.pm

```diff
@@ -0,0 +1,1408 @@
+    'guest_graphics_and_video_options'      => '',
+    'guest_serial_options'                  => '',
+    'guest_console_options'                 => '',
+    'guest_features_options'                => '',
+    'guest_power_management_options'        => '',
+    'guest_events_options'                  => '',
+    'guest_qemu_command_options'            => '',
+    'virt_install_command_line'             => '',    #This is the complete virt-install command line which is composed of above parameters end with 'options'
+    'virt_install_command_line_dryrun'      => '',    #This is [virt_install_command_line] appended with --dry-run
+    'host_ipaddr'                           => '',    #This is get_required_var('SUT_IP')
+    'host_name'                             => '',    #This is script_output('hostname')
+                                                      #[guest_os_name]:sles or others, not virt-install argument
+                                                      #[guest_os_word_length]:64 or 32, not virt-install argument
+                                                      #[guest_version]:15-sp3 or others, not virt-install argument
+                                                      #[guest_version_major]:15 or others, not virt-install argument
+                                                      #[guest_version_minor]:3 or others, not virt-intall argument
+                                                      #The following five parameters are detailed guest os information,not virt-install arguments
+    'guest_os_name'                         => '',
+    'guest_os_word_length'                  => '',
+    'guest_version'                         => '',
+    'guest_version_major'                   => '',
+    'guest_version_minor'                   => '',
+    'guest_build'                           => '',    #Build number if developing release or 'GM'
+    'guest_log_folder'                      => '',    #Log folder for individual guest [common_log_folder]/[guest_name]
+    'guest_installation_result'             => '',    #PASSED,FAILED,TIMEOUT,UNKNOWN or others
+    'guest_installation_session' => '', #Guest installation process started by "screen -t screen_title [virt_install_command_line]. It is in the form of 3401.pts-1.vh017
+    'guest_installation_session_command' => '', #If there is no [guest_installation_session] or [guest_installation_session] is terminated, start or re-connect to
+        #guest installation screen using [guest_installation_session_command] = screen -t [guest_name] virsh console --force [guest_name]
+    'guest_installation_attached' => '',    #This indicates whether guest installation screen is already connected or attached(true or false)
+    'start_run'                   => '',    #Guest creation start time
+    'stop_run'                    => ''     #Guest creation finish time
+);
+
+our $AUTOLOAD;
+our $common_log_folder           = '/tmp/guest_installations';
```

Comment:
suggest to add some key words in the log folder to differentiate this way of guest installation from traditional prj1 guest installation.